invenio_communities/cache/redis.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Implements a Redis cache."""
+from cachelib import RedisCache
+from flask import current_app
+from redis import StrictRedis
+
+from invenio_communities.cache.cache import IdentityCache
+
+
+class IdentityRedisCache(IdentityCache):
+    """Redis image cache."""
+
+    def __init__(self, app=None):
+        """Initialize the cache."""
+        super().__init__(app=app)
+        app = app or current_app
+        redis_url = app.config["COMMUNITIES_IDENTITIES_CACHE_REDIS_URL"]
+        prefix = app.config.get("COMMUNITIES_IDENTITIES_CACHE_REDIS_PREFIX", "identity")
+        self.cache = RedisCache(host=StrictRedis.from_url(redis_url), key_prefix=prefix)
+
+    def get(self, key):
+        """Return the key value.
+
+        :param key: the object's key
+        :return: the stored object
+        :rtype: `BytesIO` object
+        """
+        return self.cache.get(key)
+
+    def set(self, key, value, timeout=None):
+        """Cache the object.
+
+        :param key: the object's key
+        :param value: the stored object
+        :type value: `BytesIO` object
+        :param timeout: the cache timeout in seconds
+        """
+        timeout = timeout or self.timeout
```

Comment:
note: some keys might need to be deleted maybe? Passing `None` would be enough?

Reply:
By passing `None` there is no timeout and the keys are not deleted. you can set the value in seconds and after the set time they will be deleted. Otherwise, if we need to delete a key we can use the delete func below.